CRM/Eck/DAO/Entity.php

```diff
@@ -231,7 +231,7 @@ public static function &fields() {
         Civi::$statics[self::$_className]['fields']
       );
     }
-    return Civi::$statics[self::$_className]['fields'];
+    return Civi::$statics[self::$_className]['fields'] ?? NULL;
```

Comment:
Is it safe to return `NULL` or would it be better to return an empty array?

Reply:
At least the base class returns `NULL`, see https://github.com/civicrm/civicrm-core/blob/abfc588967b2b48469a1742d0f440ec6f6df640d/CRM/Core/DAO.php#L595

An empty array might be interpreted as *no fields* and get cached … but I'm not sure.

Reply:
The phpdoc has only `array` as return type hint so either `null` is missing or the default implementation is wrong...
How Civi treats an empty array I also currently cannot say.